test/credentials.spec.js

```diff
@@ -1474,6 +1490,48 @@
         });
       });
 
+      it('supports chaining in absense of web_identity_token_file and has source_profile ', function(done) {
+        delete process.env.AWS_WEB_IDENTITY_TOKEN_FILE;
+        helpers.spyOn(AWS.util, 'getProfilesFromSharedConfig').andReturn(
+          {
+            'source': {
+              'web_identity_token_file': 'cfgTokenFileSource',
+              'role_arn': 'cfgRoleArnSource'
```

Comment:
Can you add a test of ignoring the `source_profile` in the profile with `web_identity_token_file`? You can just add this line in this test case.

Reply:
Added test in https://github.com/aws/aws-sdk-js/pull/2837/commits/5b24b41216f8d2ee939d8b81e8a20cf21a2b4432